synthtool/languages/node.py

```diff
@@ -283,36 +283,52 @@ def owlbot_main(template_path: Optional[Path] = None):
     """
     logging.basicConfig(level=logging.DEBUG)
     # Load the default version defined in .repo-metadata.json.
-    default_version = json.load(open(".repo-metadata.json", "rt"))["default_version"]
+    default_version = json.load(open(".repo-metadata.json", "rt")).get(
+        "default_version", None
+    )
+    templates_only = json.load(open(".repo-metadata.json", "rt")).get(
+        "templates_only", None
+    )
     staging = Path("owl-bot-staging")
     s_copy = transforms.move
-    if staging.is_dir():
-        # Collect the subdirectories of the staging directory.
-        versions = [v.name for v in staging.iterdir() if v.is_dir()]
-        # Reorder the versions so the default version always comes last.
-        versions = [v for v in versions if v != default_version] + [default_version]
-
-        # Copy each version directory into the root.
-        for version in versions:
-            library = staging / version
-            _tracked_paths.add(library)
-            s_copy([library], excludes=["README.md", "package.json", "src/index.ts"])
-        # The staging directory should never be merged into the main branch.
-        shutil.rmtree(staging)
+    if templates_only:
+        # Some Node.js libraries use OwlBut purely for post-processing, see
+        # https://github.com/googleapis/nodejs-local-auth
+        common_templates = gcp.CommonTemplates(template_path)
```

Comment:
I find it hard to trace the path through this function with the nested if statements.  We could reduce the nested if depth by one by structuring it like this:

```py
if templates_only:
  versions = []
  default_version = None
elif staging.is_dir():
  ...
else:
  ...

common_templates = gcp.CommonTemplates(template_path)
...
```